src/valid/function.rs

```diff
@@ -957,7 +957,7 @@ impl super::Validator {
 
         if let Some(init) = var.init {
             let decl_ty = &gctx.types[var.ty].inner;
-            let init_ty = mod_info[init].inner_with(gctx.types);
+            let init_ty = fun_info[init].ty.inner_with(gctx.types);
```

Comment:
`var.init` is in the constant expression arena - doesn't this look up its type as if it were a local expression?